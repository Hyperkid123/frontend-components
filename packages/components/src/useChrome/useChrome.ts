import { ChromeAPI } from '@redhat-cloud-services/types';
import { useScalprum } from '@scalprum/react-core';

export type UseChromeSelector<T = any> = (chromeState: ChromeAPI) => T;

const useChrome = <T = ChromeAPI>(selector?: UseChromeSelector<T>): ChromeAPI | T => {
  const state = useScalprum<{ initialized: boolean; api: { chrome: ChromeAPI } }>();
  let chrome: ChromeAPI = state.api?.chrome || ({} as ChromeAPI);
  chrome = {
    ...chrome,
    initialized: state.initialized,
  };
  if (typeof selector === 'function') {
    return selector(chrome);
  }

  return chrome;
};

export default useChrome;
