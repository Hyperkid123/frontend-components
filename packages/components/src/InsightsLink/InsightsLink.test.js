import React from 'react';
import InsightsLink from './InsightsLink';
import useChrome from '../useChrome';
import { render } from '@testing-library/react';

jest.mock('../useChrome', () => ({
  __esModule: true,
  default: jest.fn().mockImplementation(() => ({
    isBeta: () => false,
    getApp: () => 'compliance',
    getBundle: () => 'insights',
  })),
}));
jest.mock('react-router-dom', () => ({
  Link: () => 'Mocked Link',
}));

describe('InsightsLink component', () => {
  describe('under /', () => {
    it('should render a link with a given path', () => {
      const { container } = render(
        <InsightsLink to="/scappolicies">
          Link to Scap Policies under `/insights/compliance/...` will link to `/insights/compliance/scappolicies`
        </InsightsLink>
      );
      expect(container).toMatchSnapshot();
    });

    it('should render a link to /preview with a given path', () => {
      const { container } = render(<InsightsLink to="/scappolicies" preview />);
      expect(container).toMatchSnapshot();
    });

    it('should render a link to a different app in preview when given', () => {
      const { container } = render(
        <InsightsLink to={'/ID_OF_A_SYSTEM'} app="inventory" preview>
          Link Inventory sytems page under `/insights/compliance/systems` will link to `/preview/insights/inventory/ID_OF_A_SYSTEM`
        </InsightsLink>
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe('under /preview', () => {
    beforeEach(() => {
      useChrome.mockImplementation(() => ({
        isBeta: () => true,
        getApp: () => 'compliance',
        getBundle: () => 'insights',
      }));
    });

    it('should render a link to non /preview', () => {
      const { container } = render(
        <InsightsLink to={'/ID_OF_A_SYSTEM'} app="inventory" preview={false}>
          Link Inventory sytems page under `/preview/insights/compliance/systems` will link to `/insights/inventory/ID_OF_A_SYSTEM`
        </InsightsLink>
      );
      expect(container).toMatchSnapshot();
    });

    it('should render a /preview link', () => {
      const { container } = render(
        <InsightsLink to={{ pathname: '/scappolicies' }}>
          Link to Scap Policies under `/preview/insights/compliance` will link to `/insights/compliance/scappolicies`
        </InsightsLink>
      );
      expect(container).toMatchSnapshot();
    });
  });
});
