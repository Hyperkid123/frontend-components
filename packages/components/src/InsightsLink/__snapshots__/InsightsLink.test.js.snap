// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsightsLink component under / should render a link to /preview with a given path 1`] = `
<div>
  Mocked Link
</div>
`;

exports[`InsightsLink component under / should render a link to a different app in preview when given 1`] = `
<div>
  Mocked Link
</div>
`;

exports[`InsightsLink component under / should render a link with a given path 1`] = `
<div>
  Mocked Link
</div>
`;

exports[`InsightsLink component under /preview should render a /preview link 1`] = `
<div>
  Mocked Link
</div>
`;

exports[`InsightsLink component under /preview should render a link to non /preview 1`] = `
<div>
  Mocked Link
</div>
`;
